### PR TITLE
Allow specifying HTTP basic authentication credentials when writing to Elasticsearch

### PR DESCRIPTION
**Pull request by dhruvbansal**
Useful when Elasticsearch is using X-Pack.

- Tested against Elasticsearch 5.x
- Still backwards-compatible with 2.x (ignores the extra info)

### CODE REVIEW

File: src/riemann/elasticsearch.clj [hunk 1]
```diff
@@ -41,6 +44,8 @@
   :es-index        Index name, default is \"riemann\".
   :index-suffix    Index-suffix, default is \"-yyyy.MM.dd\".
   :type            Type to send to index, default is \"event\".
+  :username        Username to authenticate with
```
Comment (jamtur01) [inline src/riemann/elasticsearch.clj]: Can you throw a `.` at the end of both of these.

File: src/riemann/elasticsearch.clj [hunk 2]
```diff
@@ -22,12 +22,15 @@
 
 (defn- post
   "POST to Elasticsearch."
-  [esindex formatted-event]
-  (let [http-options {:body (json/generate-string formatted-event)
+  [credentials esindex formatted-event]
+  (let [base-http-options {:body (json/generate-string formatted-event)
                       :content-type :json
```
Comment (mcorbin) [inline src/riemann/elasticsearch.clj]: Intentation problem.

**Comment by dhruvbansal** [comment]
OK, I made the indentation & documentation fixes you suggested and rebase'd back into my original commit (BTW this is my first use of 'git rebase'...hopefully I didn't screw it up ;)

**Comment by mcorbin** [comment]
I see 3 commits in the PR :p

For rebasing, you can :

- Use `git log` to see your last commits
- Use `git rebase -i HEAD~number_commit` where `number_commit` is the number of commits you want to modify/merge. Git rebase will open an editor (you can modify it with the GIT_EDITOR environment variable).

Example with `git rebase -i HEAD~2` :

You with have something like :
```
pick ae5a7cf t1
pick 1832c22 t2

# blablabla
```
Here, i have 2 commits, t1 and t2 (i did a first commit named t1 and a second commit named t2).

If you want to merge t2 into t1, you just write 

```
pick ae5a7cf t1
f 1832c22 t2
```
f is for `fixup`, it merge the commit with the previous one and delete his commit message. You can also do for example :

```
r ae5a7cf t1
f 1832c22 t2
```

r is for `reword`. Here, i merge the t2 commit into t1, and i can then modify the t1 commit message.
You have a lot of possible options with rebase (edit, squash, drop...) depending on what you want to do.

You can also use the commit hash instead of HEAD~number when rebasing.

This example is with 2 commits, but it's exactly the same thing with more.

Hope this can help ;)

**Pull request merged**